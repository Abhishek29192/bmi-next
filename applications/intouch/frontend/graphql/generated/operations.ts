import type * as SchemaTypes from "@bmi/intouch-api-types";

export type AddressLinesFragmentFragment = {
  readonly __typename?: "Address";
} & Pick<
  SchemaTypes.Address,
  "firstLine" | "secondLine" | "town" | "region" | "country" | "postcode"
>;

export type CompanyCertificationsFragment = {
  readonly __typename?: "Company";
} & Pick<SchemaTypes.Company, "certifications">;

export type ContactDetailsCollectionFragmentFragment = {
  readonly __typename?: "ContactDetailsCollection";
} & {
  readonly items: ReadonlyArray<
    SchemaTypes.Maybe<
      { readonly __typename?: "ContactDetails" } & Pick<
        SchemaTypes.ContactDetails,
        "fullName" | "subHeading" | "email" | "phoneNumber"
      >
    >
  >;
};

export type UpdateCompanyDetailsMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.UpdateCompanyInput;
}>;

export type UpdateCompanyDetailsMutation = {
  readonly __typename?: "Mutation";
} & {
  readonly updateCompany?: SchemaTypes.Maybe<
    { readonly __typename?: "UpdateCompanyPayload" } & {
      readonly company?: SchemaTypes.Maybe<
        { readonly __typename?: "Company" } & CompanyDetailsFragmentFragment
      >;
    }
  >;
};

export type GetGlobalDataQueryVariables = SchemaTypes.Exact<{
  [key: string]: never;
}>;

export type GetGlobalDataQuery = { readonly __typename?: "Query" } & {
  readonly marketContentCollection?: SchemaTypes.Maybe<
    { readonly __typename?: "MarketContentCollection" } & {
      readonly items: ReadonlyArray<
        SchemaTypes.Maybe<
          { readonly __typename?: "MarketContent" } & Pick<
            SchemaTypes.MarketContent,
            "externalLinkUrl" | "externalLinkLabel"
          > & {
              readonly footerLinksCollection?: SchemaTypes.Maybe<
                {
                  readonly __typename?: "MarketContentFooterLinksCollection";
                } & {
                  readonly items: ReadonlyArray<
                    SchemaTypes.Maybe<
                      { readonly __typename?: "ContentArticle" } & Pick<
                        SchemaTypes.ContentArticle,
                        "title" | "relativePath"
                      >
                    >
                  >;
                }
              >;
              readonly contactUsPage?: SchemaTypes.Maybe<
                { readonly __typename?: "ContentArticle" } & Pick<
                  SchemaTypes.ContentArticle,
                  "title" | "relativePath"
                >
              >;
            }
        >
      >;
    }
  >;
};

export type CompanyAdminsFragmentFragment = {
  readonly __typename?: "Company";
} & {
  readonly companyMembers: {
    readonly __typename?: "CompanyMembersConnection";
  } & {
    readonly nodes: ReadonlyArray<
      { readonly __typename?: "CompanyMember" } & {
        readonly account?: SchemaTypes.Maybe<
          { readonly __typename?: "Account" } & Pick<
            SchemaTypes.Account,
            | "role"
            | "id"
            | "firstName"
            | "lastName"
            | "phone"
            | "email"
            | "photo"
          >
        >;
      }
    >;
  };
};

export type InviteMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.InviteInput;
}>;

export type InviteMutation = { readonly __typename?: "Mutation" } & {
  readonly invite?: SchemaTypes.Maybe<
    ReadonlyArray<
      SchemaTypes.Maybe<
        { readonly __typename?: "Invitation" } & Pick<
          SchemaTypes.Invitation,
          "id" | "invitee"
        > & {
            readonly senderAccount?: SchemaTypes.Maybe<
              { readonly __typename?: "Account" } & Pick<
                SchemaTypes.Account,
                "email"
              >
            >;
          }
      >
    >
  >;
};

export type GetProjectQueryVariables = SchemaTypes.Exact<{
  projectId: SchemaTypes.Scalars["Int"];
}>;

export type GetProjectQuery = { readonly __typename?: "Query" } & {
  readonly project?: SchemaTypes.Maybe<
    { readonly __typename?: "Project" } & Pick<
      SchemaTypes.Project,
      | "id"
      | "name"
      | "technology"
      | "roofArea"
      | "startDate"
      | "endDate"
      | "description"
      | "buildingOwnerFirstname"
      | "buildingOwnerLastname"
      | "buildingOwnerCompany"
      | "buildingOwnerMail"
    > & {
        readonly siteAddress?: SchemaTypes.Maybe<
          { readonly __typename?: "Address" } & Pick<
            SchemaTypes.Address,
            "firstLine" | "secondLine" | "town" | "region" | "postcode"
          >
        >;
        readonly buildingOwnerAddress?: SchemaTypes.Maybe<
          { readonly __typename?: "Address" } & Pick<
            SchemaTypes.Address,
            "firstLine" | "secondLine" | "town" | "region" | "postcode"
          >
        >;
        readonly guarantees: {
          readonly __typename?: "GuaranteesConnection";
        } & {
          readonly nodes: ReadonlyArray<
            { readonly __typename?: "Guarantee" } & Pick<
              SchemaTypes.Guarantee,
              "id" | "guaranteeTypeId"
            > & {
                readonly guaranteeType?: SchemaTypes.Maybe<
                  { readonly __typename?: "ContentfulGuaranteeType" } & Pick<
                    SchemaTypes.ContentfulGuaranteeType,
                    "name"
                  > & {
                      readonly evidenceCategoriesCollection?: SchemaTypes.Maybe<
                        {
                          readonly __typename?: "ContentfulEvidenceCategoryCollection";
                        } & {
                          readonly items?: SchemaTypes.Maybe<
                            ReadonlyArray<
                              SchemaTypes.Maybe<
                                {
                                  readonly __typename?: "ContentfulEvidenceCategory";
                                } & Pick<
                                  SchemaTypes.ContentfulEvidenceCategory,
                                  "name" | "minimumUploads"
                                >
                              >
                            >
                          >;
                        }
                      >;
                    }
                >;
              }
          >;
        };
        readonly evidenceItems: {
          readonly __typename?: "EvidenceItemsConnection";
        } & {
          readonly nodes: ReadonlyArray<
            { readonly __typename?: "EvidenceItem" } & Pick<
              SchemaTypes.EvidenceItem,
              | "id"
              | "name"
              | "guaranteeId"
              | "evidenceCategoryType"
              | "customEvidenceCategoryId"
            > & {
                readonly customEvidenceCategory?: SchemaTypes.Maybe<
                  { readonly __typename?: "ContentfulEvidenceCategory" } & Pick<
                    SchemaTypes.ContentfulEvidenceCategory,
                    "name" | "minimumUploads"
                  >
                >;
              }
          >;
        };
        readonly notes: { readonly __typename?: "NotesConnection" } & {
          readonly nodes: ReadonlyArray<
            { readonly __typename?: "Note" } & Pick<
              SchemaTypes.Note,
              "id" | "body" | "createdAt"
            >
          >;
        };
      }
  >;
};

export type AccountByEmailQueryVariables = SchemaTypes.Exact<{
  email: SchemaTypes.Scalars["String"];
}>;

export type AccountByEmailQuery = { readonly __typename?: "Query" } & {
  readonly accountByEmail?: SchemaTypes.Maybe<
    { readonly __typename?: "Account" } & Pick<
      SchemaTypes.Account,
      | "id"
      | "role"
      | "marketId"
      | "firstName"
      | "lastName"
      | "email"
      | "doceboUserId"
    > & {
        readonly market?: SchemaTypes.Maybe<
          { readonly __typename?: "Market" } & Pick<
            SchemaTypes.Market,
            | "domain"
            | "language"
            | "doceboCompanyAdminBranchId"
            | "doceboInstallersBranchId"
          >
        >;
        readonly companyMembers: {
          readonly __typename?: "CompanyMembersConnection";
        } & {
          readonly nodes: ReadonlyArray<
            { readonly __typename?: "CompanyMember" } & {
              readonly company?: SchemaTypes.Maybe<
                { readonly __typename?: "Company" } & Pick<
                  SchemaTypes.Company,
                  "id" | "status" | "name"
                >
              >;
            }
          >;
        };
      }
  >;
};

export type CreateAccountMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.CreateAccountInput;
}>;

export type CreateAccountMutation = { readonly __typename?: "Mutation" } & {
  readonly createAccount?: SchemaTypes.Maybe<
    { readonly __typename?: "CreateAccountPayload" } & {
      readonly account?: SchemaTypes.Maybe<
        { readonly __typename?: "Account" } & Pick<
          SchemaTypes.Account,
          "id" | "role" | "email" | "firstName" | "lastName" | "marketId"
        > & {
            readonly market?: SchemaTypes.Maybe<
              { readonly __typename?: "Market" } & Pick<
                SchemaTypes.Market,
                | "language"
                | "doceboCompanyAdminBranchId"
                | "doceboInstallersBranchId"
              >
            >;
            readonly companyMembers: {
              readonly __typename?: "CompanyMembersConnection";
            } & {
              readonly nodes: ReadonlyArray<
                { readonly __typename?: "CompanyMember" } & {
                  readonly company?: SchemaTypes.Maybe<
                    { readonly __typename?: "Company" } & Pick<
                      SchemaTypes.Company,
                      "id" | "status"
                    >
                  >;
                }
              >;
            };
          }
      >;
    }
  >;
};

export type CreateDoceboUserMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.UserCreateInput;
}>;

export type CreateDoceboUserMutation = { readonly __typename?: "Mutation" } & {
  readonly createDoceboUser?: SchemaTypes.Maybe<
    { readonly __typename?: "UserCreateResponse" } & Pick<
      SchemaTypes.UserCreateResponse,
      "success" | "user_id"
    >
  >;
};

export type UpdateAccountMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.UpdateAccountInput;
}>;

export type UpdateAccountMutation = { readonly __typename?: "Mutation" } & {
  readonly updateAccount?: SchemaTypes.Maybe<
    { readonly __typename?: "UpdateAccountPayload" } & {
      readonly account?: SchemaTypes.Maybe<
        { readonly __typename?: "Account" } & Pick<
          SchemaTypes.Account,
          "id" | "doceboUserId"
        >
      >;
    }
  >;
};

export type UserByEmailQueryVariables = SchemaTypes.Exact<{
  email: SchemaTypes.Scalars["String"];
}>;

export type UserByEmailQuery = { readonly __typename?: "Query" } & {
  readonly userByEmail?: SchemaTypes.Maybe<
    { readonly __typename?: "UserData" } & Pick<SchemaTypes.UserData, "user_id">
  >;
};

export type InvitationsQueryVariables = SchemaTypes.Exact<{
  invitee: SchemaTypes.Scalars["String"];
}>;

export type InvitationsQuery = { readonly __typename?: "Query" } & {
  readonly invitations?: SchemaTypes.Maybe<
    { readonly __typename?: "InvitationsConnection" } & {
      readonly nodes: ReadonlyArray<
        { readonly __typename?: "Invitation" } & Pick<
          SchemaTypes.Invitation,
          "id" | "status" | "invitee" | "senderAccountId"
        >
      >;
    }
  >;
};

export type CompleteInvitationMutationVariables = SchemaTypes.Exact<{
  companyId: SchemaTypes.Scalars["Int"];
}>;

export type CompleteInvitationMutation = {
  readonly __typename?: "Mutation";
} & {
  readonly completeInvitation?: SchemaTypes.Maybe<
    { readonly __typename?: "Account" } & Pick<
      SchemaTypes.Account,
      "id" | "role" | "email" | "firstName" | "lastName" | "marketId"
    > & {
        readonly market?: SchemaTypes.Maybe<
          { readonly __typename?: "Market" } & Pick<
            SchemaTypes.Market,
            | "language"
            | "domain"
            | "doceboCompanyAdminBranchId"
            | "doceboInstallersBranchId"
          >
        >;
      }
  >;
};

export type CreateSsoUrlMutationVariables = SchemaTypes.Exact<{
  username: SchemaTypes.Scalars["String"];
  path?: SchemaTypes.Maybe<SchemaTypes.Scalars["String"]>;
}>;

export type CreateSsoUrlMutation = { readonly __typename?: "Mutation" } & {
  readonly createSSOUrl?: SchemaTypes.Maybe<
    { readonly __typename?: "SSOUrlOutput" } & Pick<
      SchemaTypes.SsoUrlOutput,
      "url"
    >
  >;
};

export type ProductsAndSystemsQueryVariables = SchemaTypes.Exact<{
  [key: string]: never;
}>;

export type ProductsAndSystemsQuery = { readonly __typename?: "Query" } & {
  readonly products?: SchemaTypes.Maybe<
    { readonly __typename?: "ProductsConnection" } & {
      readonly nodes: ReadonlyArray<
        { readonly __typename?: "Product" } & Pick<
          SchemaTypes.Product,
          | "id"
          | "bmiRef"
          | "family"
          | "name"
          | "brand"
          | "published"
          | "description"
        >
      >;
    }
  >;
  readonly systems?: SchemaTypes.Maybe<
    { readonly __typename?: "SystemsConnection" } & {
      readonly nodes: ReadonlyArray<
        { readonly __typename?: "System" } & Pick<
          SchemaTypes.System,
          "id" | "bmiRef" | "description" | "name" | "published"
        >
      >;
    }
  >;
};

export type BulkImportMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.BulkImportInput;
}>;

export type BulkImportMutation = { readonly __typename?: "Mutation" } & {
  readonly bulkImport?: SchemaTypes.Maybe<
    { readonly __typename?: "ImportPayload" } & {
      readonly systemsToInsert?: SchemaTypes.Maybe<
        ReadonlyArray<
          SchemaTypes.Maybe<
            { readonly __typename?: "System" } & Pick<
              SchemaTypes.System,
              "bmiRef"
            >
          >
        >
      >;
      readonly systemsToUpdate?: SchemaTypes.Maybe<
        ReadonlyArray<
          SchemaTypes.Maybe<
            { readonly __typename?: "System" } & Pick<
              SchemaTypes.System,
              "bmiRef"
            >
          >
        >
      >;
      readonly productsToInsert?: SchemaTypes.Maybe<
        ReadonlyArray<
          SchemaTypes.Maybe<
            { readonly __typename?: "Product" } & Pick<
              SchemaTypes.Product,
              "bmiRef"
            >
          >
        >
      >;
      readonly productsToUpdate?: SchemaTypes.Maybe<
        ReadonlyArray<
          SchemaTypes.Maybe<
            { readonly __typename?: "Product" } & Pick<
              SchemaTypes.Product,
              "bmiRef"
            >
          >
        >
      >;
    }
  >;
};

export type UpdateProductMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.UpdateProductInput;
}>;

export type UpdateProductMutation = { readonly __typename?: "Mutation" } & {
  readonly updateProduct?: SchemaTypes.Maybe<
    { readonly __typename?: "UpdateProductPayload" } & {
      readonly product?: SchemaTypes.Maybe<
        { readonly __typename?: "Product" } & Pick<SchemaTypes.Product, "id">
      >;
    }
  >;
};

export type UpdateSystemMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.UpdateSystemInput;
}>;

export type UpdateSystemMutation = { readonly __typename?: "Mutation" } & {
  readonly updateSystem?: SchemaTypes.Maybe<
    { readonly __typename?: "UpdateSystemPayload" } & {
      readonly system?: SchemaTypes.Maybe<
        { readonly __typename?: "System" } & Pick<SchemaTypes.System, "id">
      >;
    }
  >;
};

export type CreateCompanyMutationVariables = SchemaTypes.Exact<{
  input: SchemaTypes.UpdateCompanyInput;
}>;

export type CreateCompanyMutation = { readonly __typename?: "Mutation" } & {
  readonly updateCompany?: SchemaTypes.Maybe<
    { readonly __typename?: "UpdateCompanyPayload" } & {
      readonly company?: SchemaTypes.Maybe<
        { readonly __typename?: "Company" } & Pick<SchemaTypes.Company, "name">
      >;
    }
  >;
};

export type CurrentCompanyQueryVariables = SchemaTypes.Exact<{
  [key: string]: never;
}>;

export type CurrentCompanyQuery = { readonly __typename?: "Query" } & Pick<
  SchemaTypes.Query,
  "currentCompany"
>;

export type GetCurrentCompanyQueryVariables = SchemaTypes.Exact<{
  [key: string]: never;
}>;

export type GetCurrentCompanyQuery = { readonly __typename?: "Query" } & Pick<
  SchemaTypes.Query,
  "currentCompany"
>;

export type GetCompanyQueryVariables = SchemaTypes.Exact<{
  companyId: SchemaTypes.Scalars["Int"];
}>;

export type GetCompanyQuery = { readonly __typename?: "Query" } & {
  readonly company?: SchemaTypes.Maybe<
    { readonly __typename?: "Company" } & CompanyDetailsFragmentFragment
  >;
  readonly contactDetailsCollection?: SchemaTypes.Maybe<
    {
      readonly __typename?: "ContactDetailsCollection";
    } & ContactDetailsCollectionFragmentFragment
  >;
};

export type CompanyDetailsFragmentFragment = {
  readonly __typename?: "Company";
} & Pick<
  SchemaTypes.Company,
  | "id"
  | "name"
  | "logo"
  | "phone"
  | "website"
  | "aboutUs"
  | "taxNumber"
  | "tier"
  | "businessType"
  | "ownerFullname"
  | "ownerEmail"
  | "ownerPhone"
  | "publicEmail"
  | "linkedIn"
  | "facebook"
  | "referenceNumber"
> & {
    readonly tradingAddress?: SchemaTypes.Maybe<
      { readonly __typename?: "Address" } & {
        readonly coordinates?: SchemaTypes.Maybe<
          { readonly __typename?: "Point" } & Pick<SchemaTypes.Point, "x" | "y">
        >;
      } & AddressLinesFragmentFragment
    >;
    readonly registeredAddress?: SchemaTypes.Maybe<
      { readonly __typename?: "Address" } & AddressLinesFragmentFragment
    >;
  } & CompanyAdminsFragmentFragment &
  CompanyCertificationsFragment;

export type ImageFragmentFragment = { readonly __typename?: "Asset" } & Pick<
  SchemaTypes.Asset,
  | "title"
  | "description"
  | "contentType"
  | "fileName"
  | "size"
  | "url"
  | "width"
  | "height"
>;

export type GetPartnerBrandsQueryVariables = SchemaTypes.Exact<{
  [key: string]: never;
}>;

export type GetPartnerBrandsQuery = { readonly __typename?: "Query" } & {
  readonly marketContentCollection?: SchemaTypes.Maybe<
    { readonly __typename?: "MarketContentCollection" } & {
      readonly items: ReadonlyArray<
        SchemaTypes.Maybe<
          { readonly __typename?: "MarketContent" } & {
            readonly partnerBrandsCollection?: SchemaTypes.Maybe<
              {
                readonly __typename?: "MarketContentPartnerBrandsCollection";
              } & {
                readonly items: ReadonlyArray<
                  SchemaTypes.Maybe<
                    { readonly __typename?: "PartnerBrand" } & Pick<
                      SchemaTypes.PartnerBrand,
                      "name" | "shortDescription"
                    > & {
                        readonly image?: SchemaTypes.Maybe<
                          {
                            readonly __typename?: "Asset";
                          } & ImageFragmentFragment
                        >;
                        readonly logo?: SchemaTypes.Maybe<
                          {
                            readonly __typename?: "Asset";
                          } & ImageFragmentFragment
                        >;
                      }
                  >
                >;
              }
            >;
          }
        >
      >;
    }
  >;
};

export type GetProjectsQueryVariables = SchemaTypes.Exact<{
  [key: string]: never;
}>;

export type GetProjectsQuery = { readonly __typename?: "Query" } & {
  readonly projects?: SchemaTypes.Maybe<
    { readonly __typename?: "ProjectsConnection" } & {
      readonly nodes: ReadonlyArray<
        { readonly __typename?: "Project" } & Pick<
          SchemaTypes.Project,
          "id" | "name" | "technology" | "startDate" | "endDate"
        > & {
            readonly siteAddress?: SchemaTypes.Maybe<
              { readonly __typename?: "Address" } & Pick<
                SchemaTypes.Address,
                "town" | "postcode"
              >
            >;
          }
      >;
    }
  >;
};

export type CompanyMembersQueryVariables = SchemaTypes.Exact<{
  expiryDate?: SchemaTypes.Maybe<SchemaTypes.Scalars["Datetime"]>;
}>;

export type CompanyMembersQuery = { readonly __typename?: "Query" } & {
  readonly companyMembers?: SchemaTypes.Maybe<
    { readonly __typename?: "CompanyMembersConnection" } & {
      readonly nodes: ReadonlyArray<
        { readonly __typename?: "CompanyMember" } & Pick<
          SchemaTypes.CompanyMember,
          "id"
        > & {
            readonly company?: SchemaTypes.Maybe<
              { readonly __typename?: "Company" } & Pick<
                SchemaTypes.Company,
                "name"
              >
            >;
            readonly account?: SchemaTypes.Maybe<
              { readonly __typename?: "Account" } & Pick<
                SchemaTypes.Account,
                | "id"
                | "role"
                | "email"
                | "phone"
                | "photo"
                | "lastName"
                | "firstName"
              > & {
                  readonly certificationsByDoceboUserId: {
                    readonly __typename?: "CertificationsConnection";
                  } & {
                    readonly nodes: ReadonlyArray<
                      { readonly __typename?: "Certification" } & Pick<
                        SchemaTypes.Certification,
                        "id" | "name" | "technology" | "expiryDate"
                      >
                    >;
                  };
                }
            >;
          }
      >;
    }
  >;
};

export type TrainingQueryVariables = SchemaTypes.Exact<{
  catalogueId?: SchemaTypes.Maybe<SchemaTypes.Scalars["Int"]>;
  userId?: SchemaTypes.Maybe<SchemaTypes.Scalars["Int"]>;
}>;

export type TrainingQuery = { readonly __typename?: "Query" } & {
  readonly trainingContentCollection?: SchemaTypes.Maybe<
    { readonly __typename?: "TrainingContentCollection" } & {
      readonly items: ReadonlyArray<
        SchemaTypes.Maybe<
          { readonly __typename?: "TrainingContent" } & Pick<
            SchemaTypes.TrainingContent,
            | "pageHeading"
            | "description"
            | "lmsCtaLabel"
            | "pageSubHeading"
            | "step1Heading"
            | "step1SubHeading"
            | "step1Description"
            | "step2Heading"
            | "step2SubHeading"
            | "step2Description"
            | "step3Heading"
            | "step3SubHeading"
            | "step3Description"
            | "live"
          > & {
              readonly image?: SchemaTypes.Maybe<
                { readonly __typename?: "Asset" } & Pick<
                  SchemaTypes.Asset,
                  "url"
                >
              >;
            }
        >
      >;
    }
  >;
  readonly courseCatalogues?: SchemaTypes.Maybe<
    { readonly __typename?: "CourseCataloguesConnection" } & {
      readonly nodes: ReadonlyArray<
        { readonly __typename?: "CourseCatalogue" } & {
          readonly course?: SchemaTypes.Maybe<
            { readonly __typename?: "Course" } & Pick<
              SchemaTypes.Course,
              | "courseId"
              | "name"
              | "technology"
              | "image"
              | "promoted"
              | "trainingType"
              | "description"
            > & {
                readonly courseEnrollments: {
                  readonly __typename?: "CourseEnrollmentsConnection";
                } & {
                  readonly nodes: ReadonlyArray<
                    { readonly __typename?: "CourseEnrollment" } & Pick<
                      SchemaTypes.CourseEnrollment,
                      "id" | "status" | "url" | "courseId"
                    >
                  >;
                };
              }
          >;
        }
      >;
    }
  >;
};

export type DoceboCatalogIdByMarketDomainQueryVariables = SchemaTypes.Exact<{
  domain: SchemaTypes.Scalars["String"];
}>;

export type DoceboCatalogIdByMarketDomainQuery = {
  readonly __typename?: "Query";
} & {
  readonly marketByDomain?: SchemaTypes.Maybe<
    { readonly __typename?: "Market" } & Pick<
      SchemaTypes.Market,
      "doceboCatalogueId"
    >
  >;
};
