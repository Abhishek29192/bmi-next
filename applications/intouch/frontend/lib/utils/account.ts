export const randomPassword = (length: number = 8) => {
  const charset =
    "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";

  return [...new Array(length)]
    .map(() => charset.charAt(Math.floor(Math.random() * charset.length)))
    .join("");
};

export const sortByFirstName = (nodes) =>
  [...nodes].sort((a, b) => a.firstName?.localeCompare(b?.firstName));
