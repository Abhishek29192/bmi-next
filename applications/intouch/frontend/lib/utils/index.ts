import { Request } from "express";

export const formatDate = (date: string) =>
  new Intl.DateTimeFormat("en-GB", {
    dateStyle: "medium"
  } as any).format(new Date(date?.substring(0, 10)));

type MarketEnvProps = {
  currentHost: string;
  environment: string;
  market: string;
  paths: string[];
};

export const getMarketAndEnvFromReq = ({
  headers
}: Request): MarketEnvProps => {
  let market = null;
  let environment = null;

  let { host } = headers;

  const paths = host.split(".");
  const subdomain = paths[0];

  if (subdomain.indexOf("-") !== -1) {
    const parts = subdomain.split("-");
    environment = parts[0];
    market = parts[1];
  } else {
    market = subdomain;
  }

  return {
    currentHost: host,
    environment,
    market,
    paths
  };
};

export const sortArrayByField = (nodes: any[], field: string) =>
  nodes?.sort((a, b) => a?.[field]?.localeCompare(b?.[field]));
