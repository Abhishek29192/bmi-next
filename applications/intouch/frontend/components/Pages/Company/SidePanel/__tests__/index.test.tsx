import React from "react";
import { fireEvent, screen } from "@testing-library/react";
import { CompaniesSidePanel } from "..";
import { render } from "../../../../../lib/tests/utils";
import I18nProvider from "../../../../../lib/tests/fixtures/i18n";
import AccountContextWrapper from "../../../../../lib/tests/fixtures/account";
import { GetCompaniesByMarketQuery } from "../../../../../graphql/generated/operations";

describe("CompaniesSidePanel component", () => {
  const emptyNodes = {
    nodes: []
  };
  const companies: GetCompaniesByMarketQuery["companies"]["nodes"] = [
    {
      __typename: "Company",
      id: 1,
      name: "A company",
      certifications: [],
      companyDocuments: emptyNodes,
      companyMembers: emptyNodes,
      companyOperationsByCompany: emptyNodes,
      isProfileComplete: true,
      referenceNumber: 10001,
      tier: "T1",
      updatedAt: "2022-12-12"
    },
    {
      __typename: "Company",
      id: 2,
      name: "Z company",
      certifications: [],
      companyDocuments: emptyNodes,
      companyMembers: emptyNodes,
      companyOperationsByCompany: emptyNodes,
      isProfileComplete: true,
      referenceNumber: 10001,
      tier: "T1",
      updatedAt: "2021-12-12"
    }
  ];
  it("renders correctly", () => {
    const { container } = render(
      <I18nProvider>
        <AccountContextWrapper>
          <CompaniesSidePanel companies={companies} />
        </AccountContextWrapper>
      </I18nProvider>
    );
    expect(container).toMatchSnapshot();
  });
  it("check order onclick behavior", async () => {
    const { container } = render(
      <I18nProvider>
        <AccountContextWrapper>
          <CompaniesSidePanel companies={companies} />
        </AccountContextWrapper>
      </I18nProvider>
    );
    const orderListButtons = container.querySelectorAll(
      ".filterButtons > button"
    );
    fireEvent.click(orderListButtons[1]);
    expect(
      screen.getAllByTestId("companyCard")[0].querySelector("button")
    ).toHaveTextContent("A company");
  });
  it("change filter correctly", () => {
    const { container } = render(
      <I18nProvider>
        <AccountContextWrapper>
          <CompaniesSidePanel companies={companies} />
        </AccountContextWrapper>
      </I18nProvider>
    );
    fireEvent.change(container.querySelector("#filter"), {
      target: { value: "Z company" }
    });
    fireEvent.click(
      screen.getAllByTestId("companyCard")[0].querySelector("button")
    );
    expect(screen.getAllByTestId("companyCard")).toHaveLength(1);
  });
});
