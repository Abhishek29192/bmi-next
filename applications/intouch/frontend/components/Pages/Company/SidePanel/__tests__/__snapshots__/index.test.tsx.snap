// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompaniesSidePanel component renders correctly 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="sidePanel"
    >
      <div
        class="filters"
      >
        <nav>
          <div
            class="MuiFormControl-root MuiTextField-root TextField-root-2 TextField-leftAdornment-3 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="filter"
              id="filter-label"
            >
              search.inputLabel
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                id="filter"
                name="filter"
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-7"
                >
                  <span>
                    search.inputLabel
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </nav>
        <div
          class="filterButtons"
        >
          <span>
            Sort by
            :
          </span>
          <div
            class="filterButton"
          >
            <div
              class="MuiButtonBase-root MuiChip-root Chip-root-9 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-clickable"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label"
              >
                order.labels.ALPHABETICAL
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
          <div
            class="filterButton"
          >
            <div
              class="MuiButtonBase-root MuiChip-root Chip-root-9 Chip-default-10 MuiChip-clickable"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label"
              >
                order.labels.UPDATED
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="results"
      >
        <div
          data-testid="companyCard"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardContent-root main"
            >
              <h6
                class="MuiTypography-root title Typography-root-12 MuiTypography-h6"
                data-testid="typography"
              >
                A company
              </h6>
              <div>
                <p
                  class="MuiTypography-root Typography-root-12 MuiTypography-body1"
                  data-testid="typography"
                  style="display: flex;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-root"
                    focusable="false"
                    style="color: rgb(0, 154, 68);"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                    />
                  </svg>
                  search.profileComplete
                </p>
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root chevron MuiSvgIcon-colorDisabled"
                focusable="false"
                style="font-size: 36px;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          data-testid="companyCard"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardContent-root main"
            >
              <h6
                class="MuiTypography-root title Typography-root-12 MuiTypography-h6"
                data-testid="typography"
              >
                Z company
              </h6>
              <div>
                <p
                  class="MuiTypography-root Typography-root-12 MuiTypography-body1"
                  data-testid="typography"
                  style="display: flex;"
                >
                  search.profileIncomplete
                </p>
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root chevron MuiSvgIcon-colorDisabled"
                focusable="false"
                style="font-size: 36px;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="footerBtn"
      />
    </div>
  </div>
</div>
`;
